q2_vsearch/tests/test_cluster_features.py

```diff
@@ -696,6 +713,57 @@ def test_skip_denovo(self):
         exp_ref_seqs = _read_seqs(self.ref_sequences)
         self.assertEqual(obs_ref_seqs, exp_ref_seqs)
 
+    def test_skip_closed_reference(self):
+        # feature1 and feature3 clusters into r1 and feature2 and feature4
+        # clusters into r2 during closed-ref clustering; no unclustered
+        # features so de-novo clustering is skipped.
+
+        exp_table = biom.Table(np.array([[100, 101, 103],
+                                         [1, 1, 2],
+                                         [4, 5, 6],
+                                         [7, 8, 9],
+                                         [99, 98, 97]]),
+                               ['feature1', 'feature2', 'feature3',
+                                'feature4', 'feature5'],
+                               ['sample1', 'sample2', 'sample3'])
+        ref_sequences_fp = self.get_data_path('ref-sequences-2.fasta')
+        ref_sequences = Artifact.import_data('FeatureData[Sequence]',
+                                             ref_sequences_fp)
+
+        with redirected_stdio(stderr=os.devnull):
+            obs_table, rep_seqs, new_ref_seqs = self.open_reference(
+                sequences=self.input_sequences, table=self.input_table,
+                reference_sequences=ref_sequences, perc_identity=1.0)
+
+        obs_table = obs_table.view(biom.Table)
+        obs_table_ids = set(obs_table.ids(axis='observation'))
+        exp_table_ids = set(exp_table.ids(axis='observation'))
+        self.assertEqual(obs_table_ids, exp_table_ids)
+
+        # order of identifiers is important for biom.Table equality
+        obs_table = \
+            obs_table.sort_order(exp_table.ids(axis='observation'),
+                                 axis='observation')
+        self.assertEqual(obs_table, exp_table)
+
+        obs_rep_seqs = _read_seqs(rep_seqs)
+        exp_rep_seqs = [self.input_sequences_list[0],  # feature1
+                        self.input_sequences_list[4],  # feature5
+                        self.input_sequences_list[3],  # feature4
+                        self.input_sequences_list[2],  # feature3
+                        self.input_sequences_list[1]]  # feature2
+        self.assertEqual(obs_rep_seqs, exp_rep_seqs)
+
+        obs_ref_seqs = _read_seqs(new_ref_seqs)
+        # The returned "new" ref seqs should be the same as the original rep
```

Comment:
This is wrong, we still want the ref_seqs to include the full reference - that's always the case (this process can only increase the size of the reference, not decrease it).